docs/babel.config.js

```diff
@@ -1,65 +1,44 @@
-const fse = require('fs-extra');
+const getBaseConfig = require('../babel.config');
 
-const alias = {
-  '@mui/x-data-grid': '../packages/grid/x-data-grid/src',
-  '@mui/x-data-grid-generator': '../packages/grid/x-data-grid-generator/src',
-  '@mui/x-data-grid-pro': '../packages/grid/x-data-grid-pro/src',
-  '@mui/x-data-grid-premium': '../packages/grid/x-data-grid-premium/src',
-  '@mui/x-date-pickers': '../packages/x-date-pickers/src',
-  '@mui/x-date-pickers-pro': '../packages/x-date-pickers-pro/src',
-  '@mui/x-charts': '../packages/x-charts/src',
-  '@mui/x-license-pro': '../packages/x-license-pro/src',
-  '@mui/docs': '../node_modules/@mui/monorepo/packages/mui-docs/src',
-  '@mui/markdown': '../node_modules/@mui/monorepo/packages/markdown',
-  '@mui/monorepo': '../node_modules/@mui/monorepo',
-  '@mui-internal/docs-utilities': '../node_modules/@mui/monorepo/packages/docs-utilities',
-  docs: '../node_modules/@mui/monorepo/docs',
-  docsx: './',
-};
+module.exports = function getBabelConfig(api) {
+  const baseConfig = getBaseConfig(api);
 
-const { version: transformRuntimeVersion } = fse.readJSONSync(
-  require.resolve('@babel/runtime-corejs2/package.json'),
-);
+  const alias = {
+    ...getBaseConfig.getDefaultAlias(),
+    '@mui/monorepo': '../node_modules/@mui/monorepo',
+    '@mui/docs': '../node_modules/@mui/monorepo/packages/mui-docs/src',
+    docsx: './',
+  };
 
-module.exports = {
-  presets: [
-    // backport of https://github.com/zeit/next.js/pull/9511
-    ['next/babel', { 'transform-runtime': { corejs: 2, version: transformRuntimeVersion } }],
-  ],
-  plugins: [
-    [
-      'babel-plugin-transform-rename-import',
-      {
-        replacements: [{ original: '@mui/utils/macros/MuiError.macro', replacement: 'react' }],
-      },
+  return {
+    assumptions: baseConfig.assumptions,
+    presets: ['next/babel'],
+    plugins: [
+      [
+        'babel-plugin-transform-rename-import',
+        {
+          replacements: [{ original: '@mui/utils/macros/MuiError.macro', replacement: 'react' }],
+        },
+      ],
+      'babel-plugin-optimize-clsx',
+      'babel-plugin-preval',
+      [
+        'babel-plugin-module-resolver',
+        {
+          alias,
+          transformFunctions: ['require', 'require.context'],
+        },
+      ],
     ],
-    'babel-plugin-optimize-clsx',
-    // for IE 11 support
-    '@babel/plugin-transform-object-assign',
```

Comment:
IIRC, we don't support IE11 anymore.

Reply:
Depends, it's also about Safari.

Reply:
Is it? Do we support Safari <v9?
It does seem to be supported [starting v9](https://caniuse.com/mdn-javascript_builtins_object_assign) on Safari and iOS Safari. 🤔